testing/README.md

```diff
@@ -0,0 +1,355 @@
+        assert.calledWith(spy, 'some arg');
+      });
+    });
+  });
+  ```
+
+- Use the most meaningful assertion for what you are trying to test. For a list of available assertions, make sure to check out [chai’s `assert` docs](http://chaijs.com/api/assert/).
+
+  > Why? More specific assertions have more specific error messages, making it easier for you (and any developer that deals with this code in the future) to track down and fix test failures.
+
+  ```js
+  // bad
+  assert.isTrue(foo === bar);
+
+  // good
+  assert.equal(foo, bar);
+  ```
+
+- Use sinon’s custom spy assertions, exposed to chai’s `assert` object.
+
+  ```js
+  // somewhere in the test setup...
+  // (see http://sinonjs.org/docs/#assert-expose)
+  sinon.assert.expose(assert);
+
+  // bad
+  assert.isTrue(spy.called);
+
+  // good
+  assert.called(spy);
+  ```
+
+- Always use a test helper that sets up your global test state. This generally includes adding any chai plugins you need, setting up a global sinon sandbox, and any other global hooks you require. It is often useful to expose `sinon`/ `sandbox` and `assert` to the global object in this helper to avoid needing to import them in every test file.
+
+  ```js
+  // Here's an example test-helper file
+
+  import chai, {assert} from 'chai';
+  import sinon from 'sinon';
+  import chaiAsPromised from 'chai-as-promised';
+
+  chai.use(chaiAsPromised);
+  sinon.assert.expose(assert);
+
+  // This creats a sandbox that will automatically restore any
+  // stubbed properties performed during tests.
+  const sandbox = sinon.sandbox.create();
+
+  global.assert = assert;
+  global.sinon = sandbox;
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+  ```
+
+- Use mocha’s ability to automatically finish tests when a returned promise is resolved for tests that are asynchronous and depend on promises.
+
+  ```js
+  // bad
+  test('it resolves with a value', (done) => {
+    doSomethingAsync()
+      .then((result) => {
+        assert.equal(result, 'something');
+        done();
+      })
+      .catch(() => {
+        assert(false, 'The promise was rejected :(');
+      });
+  });
+
+  // good
+  test('it resolves with a value', () => {
+    return doSomethingAsync()
+      .then((result) => {
+        assert.equal(result, 'something');
+      });
+  });
+
+  // also good, if using async/ await
+  test('it resolves with a value', async () => {
+    const result = await doSomethingAsync();
+    assert.equal(result, 'something');
+  });
+  ```
+
+[↑ scrollTo('#table-of-contents')](#table-of-contents)
+
+
+
+## Best practices
+
+- Try to limit the number of assertions made in a single test. For unit tests, you should only be testing one thing at a time (generally, an individual function, either standalone or as a method of an object), and more than one or two assertions is generally indicative that you should split your test into multiple tests.
+
+- Something something about how to organize suites (by method? by context?)
```

Comment:
Similar to question above — do you use nested suites for describing methods? For describing different arguments? Not at all?